chipsec/helper/linuxnative/linuxnativehelper.py

```diff
@@ -0,0 +1,772 @@
+from chipsec.logger import logger
+import chipsec.file
+from chipsec.hal.uefi_common import EFI_VARIABLE_NON_VOLATILE, EFI_VARIABLE_BOOTSERVICE_ACCESS
+from chipsec.hal.uefi_common import EFI_VARIABLE_HARDWARE_ERROR_RECORD, EFI_VARIABLE_AUTHENTICATED_WRITE_ACCESS
+from chipsec.hal.uefi_common import EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS, EFI_VARIABLE_APPEND_WRITE
+from chipsec.hal.uefi_common import EFI_VARIABLE_RUNTIME_ACCESS
+
+LZMA = os.path.join(chipsec.file.get_main_dir(), chipsec.file.TOOLS_DIR, "compression", "bin", "LzmaCompress")
+TIANO = os.path.join(chipsec.file.get_main_dir(), chipsec.file.TOOLS_DIR, "compression", "bin", "TianoCompress")
+EFI = os.path.join(chipsec.file.get_main_dir(), chipsec.file.TOOLS_DIR, "compression", "bin", "TianoCompress")
+BROTLI = os.path.join(chipsec.file.get_main_dir(), chipsec.file.TOOLS_DIR, "compression", "bin", "Brotli")
+
+_tools = {
+    chipsec.defines.COMPRESSION_TYPE_TIANO: 'TianoCompress',
+    chipsec.defines.COMPRESSION_TYPE_LZMA: 'LzmaCompress',
+    chipsec.defines.COMPRESSION_TYPE_BROTLI: 'Brotli'
+}
+
+
+class MemoryMapping(mmap.mmap):
+    """Memory mapping based on Python's mmap.
+
+    This subclass keeps tracks of the start and end of the mapping.
+    """
+    def __init__(self, fileno, length, flags, prot, offset):
+        self.start = offset
+        self.end = offset + length
+        super().__init__()
+
+
+class LinuxNativeHelper(Helper):
+
+    DEV_MEM = "/dev/mem"
+    DEV_PORT = "/dev/port"
+
+    decompression_oder_type1 = [chipsec.defines.COMPRESSION_TYPE_TIANO, chipsec.defines.COMPRESSION_TYPE_UEFI]
+    decompression_oder_type2 = [chipsec.defines.COMPRESSION_TYPE_TIANO,
+                                chipsec.defines.COMPRESSION_TYPE_UEFI,
+                                chipsec.defines.COMPRESSION_TYPE_LZMA,
+                                chipsec.defines.COMPRESSION_TYPE_BROTLI]
+
+    def __init__(self):
+        super(LinuxNativeHelper, self).__init__()
+        self.os_system = platform.system()
+        self.os_release = platform.release()
+        self.os_version = platform.version()
+        self.os_machine = platform.machine()
+        self.os_uname = platform.uname()
+        self.name = "LinuxNativeHelper"
+        self.dev_fh = None
+        self.dev_mem = None
+        self.dev_port = None
+        self.dev_msr = None
+
+        # A list of all the mappings allocated via map_io_space. When using
+        # read/write MMIO, if the region is already mapped in the process's
+        # memory, simply read/write from there.
+        self.mappings = []
+
+###############################################################################################
+# Driver/service management functions
+###############################################################################################
+    def create(self, start_driver):
+        if logger().DEBUG:
+            logger().log("[helper] Linux Helper created")
```

Comment:
Note: This will need to get cleaned up to use the new logger interfaces at some point.  Not needed to merge this time.